common/lib/xmodule/xmodule/tests/test_sequence.py

```diff
@@ -181,6 +185,51 @@ def test_timed_exam_gating_waffle_flag(self, mocked_user):
             )
             mocked_user.assert_called_once()
 
+    @override_waffle_flag(TIMED_EXAM_GATING_WAFFLE_FLAG, active=True)
+    @patch('xmodule.seq_module.User.objects.get', return_value=UserFactory.build())
+    @patch('openedx.features.content_type_gating.models.ContentTypeGatingConfig.enabled_for_enrollment',
+           return_value=True)
+    def test_that_timed_sequence_gating_respects_access_configurations(self, mocked_user, mocked_config):  # pylint: disable=unused-argument
+        """
+        Verify that if a time limited sequence contains content type gated problems, we gate the sequence
+        Verify that if a time limited sequence contains gated problems, but not due to content type gating,
+        then sequence is not gated
+        Verify that if all problems in a time limited sequence can be accessed, the sequence is not gated
+        """
+        # the one problem in this sequence needs to have graded set to true in order to test content type gating
+        self.sequence_5_1.get_children()[0].get_children()[0].graded = True
+        gated_fragment = Fragment('i_am_gated')
+
+        # When a time limited sequence contains content type gated problems, the sequence itself is gated
+        content_gating_error = AccessResponse(False, error_code="incorrect_user_group", user_fragment=gated_fragment)
+        with patch.object(SequenceModule, '_has_access_check', return_value=content_gating_error):
+            view = self._get_rendered_view(
+                self.sequence_5_1,
+                extra_context=dict(next_url='NextSequential', prev_url='PrevSequential'),
+                view=STUDENT_VIEW
+            )
+            self.assertIn('i_am_gated', view)
```

Comment:
Can you reuse `def _assert_gated`?

Reply:
no, this relates to gating for prerequisites which is not related to content type gating

Reply:
Ah ok, it looked like some of the asserts in there can be reused for these cases, esp. if we're taking the time to set up the entire course for each test.

Reply:
the test is testing specifically the gating functionality, which should be covered by the assert 
are you referencing any asserts in particular that are relevant to showing that_timed_sequence_gating_respects_access_configurations?

Reply:
```
        self.assertIn("seq_module.html", html)
        self.assertIn("'banner_text': None", html)
        self.assertIn("'gated': False", html)
        self.assertEqual('NextSequential', html['next_url'])
        self.assertEqual('PrevSequential', html['prev_url'])
```

Reply:
how are those relevant to showing that_timed_sequence_gating_respects_access_configurations?
(also the gated and banner_text are relevant to the prerequisite gating that is using those functions)

Reply:
It would fail `_assert_gated()` because `['gated_content']['gated']` is not True

Reply:
I was reading the asserts as "these are the elements on the page that describe what an ungated or gated page looks like". Since we're testing that the page renders correctly depending on the cases, the thought was that we'd want to check that the gated page renders correctly/does have a banner/can access the previous and next buttons, and that the ungated page does not have a banner/cannot access the next button, etc.

Reply:
both the gated and ungated page can access the next and previous buttons (for both this gating and the prerequisiting gating)
the asserts are specific to the prerequisite gated page which does not look the same as a page gated in this case
that said, if there are specific key elements of the page we need to test for this case, we could certainly add them to these asserts

Reply:
ah ok, it would be good to add a couple of asserts that could confirm that the correct pageview is loaded. Not familiar enough to know what those would be though (maybe banner/no banner?)

Reply:
added